### PR TITLE
`not`: Dont treat assignments as binary expression globally and skip unnecesary scopes

### LINKED ISSUE #108: Skip unnecessary scopes for `.not`
I was thinking about this for a long time. I don't think `.not` should suggest to invert identifier in this locations:

```ts
// simply invalid
if (tab.input instanceof TabInputText) continue
// suggesting to invert b here is annoying for me for a long time
if (a === b) continue

// if (a === !b) continue TS allows this only if both sides are of type boolean, but again, why don't just use `!==` ? 
```

### LINKED ISSUE #109: no `new` builtin postfix within arrow callback
```ts
fn = () => {
    throw b.new
}
```
or just
```ts
fn = () => {
    b.new
}
```

### PR DESCRIPTION
**Pull request by zardoy**
fixes https://github.com/ipatalas/vscode-postfix-ts/issues/109
fixes https://github.com/ipatalas/vscode-postfix-ts/issues/108#issuecomment-1429719522

partially fixes https://github.com/ipatalas/vscode-postfix-ts/issues/108 (at least for all of my cases)

### CODE REVIEW

File: src/postfixCompletionProvider.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export class PostfixCompletionProvider implements vsc.CompletionItemProvider {
   }
 
   private isTypeReference = (node: ts.Node) => {
-    const typeRef = findClosestParent(node, ts.SyntaxKind.TypeReference)
+    const typeRef = ts.findAncestor(node, ts.isTypeReferenceNode)
```
Comment (ipatalas) [inline src/postfixCompletionProvider.ts:78]: Ha! This is cool, somehow I never found this.

File: src/templates/notTemplate.ts [hunk 2]
```diff
@@ -43,6 +43,14 @@ export class NotTemplate extends BaseTemplate {
         || this.isIdentifier(node))
   }
 
+  private isStrictEqualityOrInstanceofBinaryEpxression = (node: ts.Node) => {
```
Comment (ipatalas) [inline src/templates/notTemplate.ts]:
```suggestion
  private isStrictEqualityOrInstanceofBinaryExpression = (node: ts.Node) => {

```

Comment (ipatalas) [reply src/templates/notTemplate.ts]: Typo

**Review by ipatalas — APPROVED** [review]
Good job (again)! 
I think you forgot to adjust two template usage tests, they are failing but this should be easy.

**Comment by zardoy** [comment]
@ipatalas done. Didn't do it on purpose as wanted to know why didn't you add `new` in these cases, but forgot to write another message to reach out you

**Comment by ipatalas** [comment]
> @ipatalas done. Didn't do it on purpose as wanted to know why didn't you add `new` in these cases, but forgot to write another message to reach out you

Oh, ok. Well, quite frankly I don't remember. Now when I look at it it looks reasonable to include it there too so perhaps I just overlooked that back then.

**Comment by zardoy** [comment]
Thanks!

**Pull request merged**